cmd/root.go

```diff
@@ -97,3 +101,38 @@ func initConfig() {
 		//	fmt.Fprintln(os.Stderr, "Using config file:", viper.ConfigFileUsed())
 	}
 }
+
+func getConfigFilePath() string {
+	return filepath.Join(xdg.ConfigHome, "k8sgpt", "k8sgpt.yaml")
+}
+
+func ensureDirExists(dir string) {
+	err := os.Mkdir(dir, 0755)
+
+	if errors.Is(err, os.ErrExist) {
+		return
+	}
+}
+
+// tests if a file exists in the legacy config path and warns the user about it
+func checkForOldConfig() {
+	home, err := os.UserHomeDir()
+
+	// ignoring the error here, as we do not want to exit here even on error
+	if err != nil {
+		return
+	}
+
+	oldConfig := filepath.Join(home, ".k8sgpt.yaml")
+
+	if _, err := os.Stat(oldConfig); err == nil {
```

Comment:
Just a suggestion for readability, to keep the "happy path" of the function outside of the if statement and possibly return a boolean and an error if old config was found when function is called.
So `checkForOldConfig` is not producing any fmt output and is only checking if config exists.

Reply:
Yes this function is quite hard to read +1

Reply:
I've replaced this whole function with `performConfigMigrationIfNeeded`. This function will test if the legacy config
exists and will react accordingly.